src/index.ts

```diff
@@ -452,21 +467,31 @@ export default class Wallet {
    * @param input A JSON serialized string, or an object representing EthSale Keystore.
    * @param password The keystore password.
    */
-  public static fromEthSale(input: string | EthSaleKeystore, password: string): Wallet {
+  public static async fromEthSale(
+    input: string | EthSaleKeystore,
+    password: string
+  ): Promise<Wallet> {
     const json: EthSaleKeystore = typeof input === 'object' ? input : JSON.parse(input)
 
     const encseed = Buffer.from(json.encseed, 'hex')
 
     // key derivation
-    const derivedKey = crypto.pbkdf2Sync(password, password, 2000, 32, 'sha256').slice(0, 16)
+    const pass = Buffer.from(password, 'utf8')
+    const derivedKey = pbkdf2Sync(pass, pass, 2000, 32, 'sha256').slice(0, 16)
 
     // seed decoding (IV is first 16 bytes)
     // NOTE: crypto (derived from openssl) when used with aes-*-cbc will handle PKCS#7 padding internally
     //       see also http://stackoverflow.com/a/31614770/4964819
-    const decipher = crypto.createDecipheriv('aes-128-cbc', derivedKey, encseed.slice(0, 16))
-    const seed = runCipherBuffer(decipher, encseed.slice(16))
+    const seed = await aes.decrypt(
+      encseed.slice(16),
+      derivedKey,
+      encseed.slice(0, 16),
+      'aes-128-cbc',
+      true
+    )
+    // const seed = runCipherBuffer(decipher, encseed.slice(16))
 
-    const wallet = new Wallet(keccak256(seed))
+    const wallet = new Wallet(Buffer.from(keccak256(Buffer.from(seed))))
```

Comment:
What does the interior `Buffer.from(seed)` accomplish here since `seed` is already a `Uint8array`?

Reply:
<img width="763" alt="Screen Shot 2022-09-08 at 5 25 29 PM" src="https://user-images.githubusercontent.com/574696/189162374-20df3732-e6e6-46c1-a3bc-e58b413f7734.png">